src/feature/control/control_events.c

```diff
@@ -1318,6 +1318,10 @@ control_event_logmsg(int severity, log_domain_mask_t domain, const char *msg)
       for (cp = b; *cp; ++cp)
         if (*cp == '\r' || *cp == '\n')
           *cp = ' ';
+
+      /* Remove trailing spaces */
+      for (--cp; *cp == ' '; --cp)
```

Comment:
Please don't write past the beginning of a string that only has spaces in it:
```suggestion
      for (--cp; *cp == ' ' && cp >= b; --cp)
```

Reply:
Agreed;
In such case, should an empty string be sent (send_control_event(...)) at all?

Reply:
That's a good question.

If we're sending a log message that is just spaces, that's a bug. But we can't log from inside a logging function, unless we disable controller logs:
`
    ++disable_log_messages;
    tor_assert_nonfatal(*b);
    --disable_log_messages;
`